dht.go

```diff
@@ -373,11 +374,21 @@ func makeDHT(h host.Host, cfg dhtcfg.Config) (*IpfsDHT, error) {
 	return dht, nil
 }
 
-// lookupCheck performs a lookup request to a remote peer.ID, verifying that it is able to
+// lookupCheck performs a random lookup request to a remote peer.ID, verifying that it is able to
 // answer it correctly
 func (dht *IpfsDHT) lookupCheck(ctx context.Context, p peer.ID) error {
-	// lookup request to p requesting for its own peer.ID
-	peerids, err := dht.protoMessenger.GetClosestPeers(ctx, p, p)
+	_, public, err := crypto.GenerateKeyPair(crypto.Ed25519, 0)
```

Comment:
This an expensive CPU operation the bits argument is ignored for Ed25519:
https://github.com/libp2p/go-libp2p/blob/cbfd89676cb2e775441d158fb607d66a94241ae2/core/crypto/key.go#L112-L113

Alternatives:
- Do it once at init (meh but fine)
- Do it once in new and store it as a field
- Use our own Peer ID
- Use a hardcoded whatever Peer ID.

Reply:
> Use our own Peer ID

👍🏻  Let's use our own Peer ID.